pytorch_lightning/trainer/trainer.py

```diff
@@ -262,11 +262,15 @@ def __init__(
 
             max_epochs: Stop training once this number of epochs is reached. Disabled by default (None).
                 If both max_epochs and max_steps are not specified, defaults to ``max_epochs`` = 1000.
+                To disable automatic stopping, you can set ``max_epochs = -1`` and set ``max_steps`` as ``None``
+                or ``-1``. Note that if the the ``max_time`` limit is specified, it will still be observed.
 
             min_epochs: Force training for at least these many epochs. Disabled by default (None).
                 If both min_epochs and min_steps are not specified, defaults to ``min_epochs`` = 1.
 
-            max_steps: Stop training after this number of steps. Disabled by default (None).
+            max_steps: Stop training after this number of steps. Disabled by default (None). If ``max_steps = None``
+                and ``max_epochs = None``, will default to ``max_epochs = 1000``. To override this
+                behavior, see ``max_epochs``.
 
```

Comment:
nit: it feels odd to mention `max_epochs` and `max_steps` in each others documentation.

Reply:
Done